glyphs-reader/src/font.rs

```diff
@@ -30,6 +30,7 @@ pub struct Font {
     pub family_name: String,
     pub axes: Option<Vec<Axis>>,
     pub glyphs: Vec<Glyph>,
+    pub glyph_order: Option<Vec<String>>,
```

Comment:
technically in the Glyphs.app API, GSFont doesn't have such 'glyph_order' or 'glyphOrder' attribute, but maybe unlike in glyphsLib/classes.py, we don't care about matching that.

Reply:
idea: instead of storing an additional `glyph_order` Vec<String> (which also doesn't exist in the Glyphs.app API), you could reorder the font.glyphs Vec<Glyph> in-place such that it matches any glyphOrder custom parameter 

(and that they do _not_ match looks unlikely given https://github.com/googlefonts/fontmake-rs/pull/43#discussion_r1045790918)

Reply:
I worry a little that maybe I just didn't find the way to make them not match but it does in fact exist.

IMHO the correct typing would be for glyphs to be a name:glyph map and glyph order to be it's own thing - which is what it turns into in IR - but we're limited wrt what the plist reader will readily accomodate.

Re-organizing the exisitng field feels all to easy to miss to me. Maybe the right choice would be to turn the raw Font from plist into another type and compute the derivative information when we transition.

Reply:
To my understanding we don't care about matching GSFont, we just want the information to power IR?

I'm going to try the two types approach.

Reply:
I have now split RawFont, what we parse from plist, and Font which has some post-processing done on it. Font now has a name:glyph map and a glyph order as separate constructs.

Reply:
I think it's better so resolving but plz reopen if you disagree